src/requestManager.cpp

```diff
@@ -587,6 +667,31 @@ void CRequestManager::SendRequests()
             }
         }
     }
+    // send batched requests if any.
+    if (fBatchBlockRequests && !mapBatchBlockRequests.empty())
+    {
+        for (auto iter : mapBatchBlockRequests)
+        {
+            LEAVE_CRITICAL_SECTION(cs_objDownloader);
+            {
+                LOCK(cs_main);
+                for (auto &inv : iter.second)
+                {
+                    MarkBlockAsInFlight(iter.first->GetId(), inv.hash, Params().GetConsensus());
+                }
+            }
+            iter.first->PushMessage(NetMsgType::GETDATA, iter.second);
+            ENTER_CRITICAL_SECTION(cs_objDownloader);
+            LOG(REQ, "Sent batched request with %d blocks to node %s\n", iter.second.size(), iter.first->GetLogName());
+        }
+
+        LOCK(cs_vNodes);
```

Comment:
This lock is not be necessary I think.  Since we have ownership of the node, it can't be removed from cs_vNodes